tingbot/graphics.py

```diff
@@ -217,6 +217,77 @@ def text(self, string, xy=None, color='grey', align='center', font=None, font_si
+                    # as long as the first two are valid numbers.
+                    if not(isinstance(xy[0], numbers.Real) and isinstance(xy[1], numbers.Real)):
+                        raise TypeError('xy should contain two numbers')
+        
+        # if size is not none
+        if size is not None:
+            # it has to contain at least two values:
+            if not(hasattr(size, '__len__')):
+                raise TypeError('size should contain at least two numbers')
+            # it can contain more than two values 
+            elif len(size) >= 2:
+                # as long as the first two are valid numbers
+                if not(isinstance(size[0], numbers.Real) and isinstance(size[0], numbers.Real)):
+                    raise TypeError('size should contain two numbers')
+                # and positive
+                elif size[0] < 0 or size[1] < 0:
+                    raise TypeError('the first two elements of size should be positive numbers')
+                # all checks passed, we need to extract the first two elements of size to draw the shape
+                else:
+                    xy = _topleft_from_aligned_xy(xy, align, (size[0:2]), self.size)
+                    pygame.draw.ellipse(self.surface, _color(color), pygame.Rect(xy, size[0:2]), 0) 
+            else:
+                xy = _topleft_from_aligned_xy(xy, align, size, self.size)
+                pygame.draw.ellipse(self.surface, _color(color), pygame.Rect(xy, size), 0) 
+
+    def circle(self, xy=None, size=100, color='grey', align='center'):
+        """
+        Draws a circle.
+
+        Args:
+            xy (tuple): The position (x, y) to draw the circle, as measured from the top-left.
+            size (int): The diameter of the circle.
+            color (tuple or str): The color (r, g, b) or color name.
+            align (str): How to align the outside box of the circle relative to `xy`, or relative to the drawing surface
+                if `xy` is None. Defaults to 'center'.
+        """
+        # if size is not none
+        if size is not None:
+            # it can contain more than one value 
+            if hasattr(size, '__len__'):
+                # if it does contain more than one
+                if len(size) >= 1:
+                    # it is ok as long as the first one is a valid number.
+                    if not isinstance(size[0], numbers.Real):
+                        raise TypeError('size should contain a number')
+                    elif size[0] < 0:
+                        raise TypeError('the first element of size should be a positive number')
+                    else:
+                        # we have to only send the first element of size twice in this case to the oval method.
+                        self.oval(xy,(size[0],size[0]), color, align)
+            else:
+                # a circle is just a special case of an oval shape:
+                self.oval(xy, (size,size), color, align)
```

Comment:
here, you can simplify the code by reassigning to the size variable to avoid the repeating self.oval line. So we can change the above code to:

```
if isinstance(size, numbers.Real):
    size = (size, size)
elif hasattr(size, '__len__') and len(size) == 2:
    pass
else:
    raise TypeError('size should be a number or a 2-tuple')

self.oval(xy=xy, size=size, color=color, align=align)
```